### PR TITLE
core(lighthouse-logger): convert to ES modules

### PR DESCRIPTION
**Pull request by connorjclark**
ref #12689

This is not a breaking change because only conversions to ESM inside `lighthouse-core` would be.

The only impact is that the `build-bundle` script will see this file as ESM during rollup (FYI: when bundling we use the local copy, but via Node we use the npm pacakge). I have no plans to publish a new version of the npm package, but if we did, that shouldn't be a breaking change for LH consumers either if LH used that new ESM version.

### CODE REVIEW

**Comment by connorjclark** [comment]
>  I have no plans to publish a new version of the npm package, but if we did, that shouldn't be a breaking change for LH consumers either if LH used that new ESM version.

This only holds for usage of LH directly.... if a user was importing the logger package themselves then of course the breaking change would be they can't `require` it anymore. By punting the publish we don't have to ask ourselves what our versioning policy should be there :)

**Comment by connorjclark** [comment]
I think this needs to land first #13722

File: lighthouse-logger/index.js [hunk 1]
```diff
@@ -5,11 +5,11 @@
  */
 'use strict';
 
-const process = require('process');
-const debug = require('debug');
-const marky = require('marky');
+import process from 'process';
+import debug from 'debug';
+import * as marky from 'marky';
```
Comment (rayjenn) [inline lighthouse-logger/index.js:10]: why so cheap

**Comment by dlockhart** [comment]
@connorjclark, @adamraine: Now that this was released ~2 hours ago, it's started blowing up. Specifically this usage in `chrome-launcher`:
https://github.com/GoogleChrome/chrome-launcher/blob/main/src/chrome-finder.ts#L13

**Comment by adamraine** [comment]
@dlockhart thanks, yeah we are working on a fix

**Pull request merged**